src/message.js

```diff
@@ -462,7 +465,10 @@ export class Message {
       if (privateKey.isPublic()) {
         throw new Error('Need private key for signing');
       }
-      const signingKey = await privateKey.getSigningKey(undefined, date, userIds);
+      const signingKeyId = privateKeys.length - 1 - i >= 0 ?
+        signingKeyIds[privateKeys.length - 1 - i] :
+        signingKeyIds[0];
+      const signingKey = await privateKey.getSigningKey(signingKeyId, date, userIds);
```

Comment:
I'm confused, shouldn't this just be `signingKeyIds[i]`?

Reply:
The array of `privateKeys` is reversed before the loop is executed, so we have to get the opposite index or also reverse the signingKeyIds array

Reply:
Ah, I see. Yeah, I'd be tempted to do something like
```js
signingKeyIds.length = privateKeys.length;
signingKeyIds.reverse();
```
beforehand, that way the symmetry between the two arrays is a bit more obvious.

Reply:
Would you suggest leaving a comment above it to aid the reader?

Reply:
I think it's fine, given that the other `reverse` is right below it, the connection should be obvious enough. If you want to make it even more clear, you could do
```js
privateKeys = Array.from(privateKeys).reverse();
signingKeyIds = Array.from(signingKeyIds);
signingKeyIds.length = privateKeys.length;
signingKeyIds.reverse();
```
(also to avoid mutating the original parameter) - and then use those variables after.

Reply:
I would not prefer to go with that path as it's an additional unnecessary step (although it's a very cheap call). Okay to proceed ?

Reply:
Ah, sure :+1:
Though then still, I think this should default to `undefined`, rather than `signingKeyIds[0]`, right? Or simply
```suggestion
      const signingKeyId = signingKeyIds[privateKeys.length - 1 - i];
      const signingKey = await privateKey.getSigningKey(signingKeyId, date, userIds);
```
should also work?

Reply:
Yes, this will work. This must have slipped through when I was running the test and forgot to build beforehand ✌️